doc/usage.rst

```diff
@@ -39,6 +39,30 @@ of ``target_link_libraries`` with a customized version::
   include(CeleritasLibrary)
   celeritas_target_link_libraries(mycode PUBLIC Celeritas::celeritas)
 
+As ``celeritas_target_link_libraries`` decays to ``target_link_libraries`` if
+CUDA and VecGeom are not used, you can use it safely to link nearly all targets consuming
+Celeritas, or Celeritas consumers, in your project to track the appropriate sequence
+of linking for the final application whether CPU-only or CUDA enabled::
+
+  add_library(myconsumer SHARED ...)
+  celeritas_target_link_libraries(myconsumer PUBLIC Celeritas::celeritas)
+
+  add_executable(myapplication ...)
+  celeritas_target_link_libraries(myapplication PRIVATE myconsumer)
+
+The one exception to this is when your project builds a shared library that will
```

Comment:
I think with the fix this is no longer needed.

Reply:
If you use a "SHARED" library as a plugin I would still use the `celeritas_` library as it would still properly handle if that library contains cuda code (i.e. the text 'works' only if the library does not contain any cuda code).

The only to note is that they should then load the "_final" library as their plugin.

Reply:
The case I was thinking about here is if a project is providing a custom `add_library` of their own, e.g. `myprojects_add_library` and similar. The secondary issue is that certain systems may enforce a naming convention for plugins, and so loading the `_final` library may not be possible without a softlink or similar. I can clarify both of these.

Reply:
Technically they should change `myprojects_add_library` to use `celeritas_add_library` :) but fair enough it makes sense and the hints is useful anyway (if only to talk about the `_final` library.